cnf-certification-test/lifecycle/suite.go

```diff
@@ -102,13 +102,13 @@ var _ = ginkgo.Describe(common.LifecycleTestKey, func() {
 		if env.GetWorkerCount() < minWorkerNodesForLifecycle {
 			ginkgo.Skip("Skipping pod scheduling test because invalid number of available workers.")
 		}
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.NonGuaranteedPods)
+		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.GetNonGuaranteedPodsWithAntiAffinity())
```

Comment:
Hi Brandon, just comparing the job executed for [this change](https://www.distributed-ci.io/jobs/1baa8941-bab7-4ad7-b0d6-d76a49a73caf/tests) and another from current [main branch](https://www.distributed-ci.io/jobs/0e57e2ec-fa49-4c26-b68f-90d6e4643ee4/tests), I saw `lifecycle-pod-scheduling` test is now being skipped. Is it because of this change, that now considers that the test is only executed for non guaranteed pods (in terms of QoS), but also having anti affinity rules?

Reply:
So I think the `main` branch run is only passing because there are no pods to test against.  The change here tells the code to skip if there are no pods that meet those criteria.  I think the main branch should probably "skip" as well but we only skip based on these TestEnvironment groupings being empty prior to executing the test.

Reply:
Okey, got it! Just to know that it was expected to do that change. I'll test it with a new change I have included in our example tested by tnf: https://softwarefactory-project.io/r/c/dci-openshift-app-agent/+/26228, just to check that, effectively, a non guaranteed pod with affinity rule enabled is tested. You'll see the results in future DCI jobs launched under this PR.